scripts/zones/Windurst_Waters_[S]/IDs.lua

```diff
@@ -42,10 +43,24 @@ zones[xi.zone.WINDURST_WATERS_S] =
     },
     mob =
     {
+        CAMPAIGN =
+        {
+            KAYEEL_PAYEEL           = GetFirstID('Kayeel-Payeel'),           -- +1 Ice_Fiend, +2 Thunder_Fiend, +3-11 Aquarian_Caster
+            GARIRI                  = GetFirstID('Gariri'),                  -- +1 Flame_Giant, +2-10 Scorpion_Caster
+            ZOLKU_AZOLKU            = GetFirstID('Zolku-Azolku'),            -- +1-9 Libran_Caster
+            LHU_MHAKARACCA          = GetFirstID('Lhu_Mhakaracca'),          -- +1 Pya, +2 Kyo, +3-11 Python_Mercenary
+            HAJA_ZHWAN              = GetFirstID('Haja_Zhwan'),              -- +1-9 Wildcat_Volunteer
+            VEE_QIQA_THE_DECREER    = GetFirstID('Vee_Qiqa_the_Decreer'),    -- +1-4 Vee_Qiqas_Elemental, +5-13 Divine_Templar
+            MOO_OUZI_THE_SWIFTBLADE = GetFirstID('Moo_Ouzi_the_Swiftblade'), -- +9 Divine_Disseminator
+            MUU_BUXU_THE_ELUSIVE    = GetFirstID('Muu_Buxu_the_Elusive'),    -- +9 Divine_Assassin
+            DEE_XALMO_THE_GRIM      = GetFirstID('Dee_Xalmo_the_Grim'),      -- +9 Divine_Ascetic
+            KAZAN_THE_PEERLESS      = GetFirstID('Kazan_the_Peerless'),      -- +9 Plenilune_Ronin
+        },
```

Comment:
Given that these are the exact same blocks repeated in all of these ID files, it smells like there might be a better design. It might be worth populating them elsewhere (a campaign global?) with `:queryEntitiesByName(...)`, since that uses the same/similar caching/lookup mechanisms to this and similarly incurs no performance penalty

Reply:
we have https://github.com/LandSandBoat/server/blob/base/scripts/globals/campaign.lua#L34 and this feels small enough that we wouldn't need a 2nd, so we can probably add this about line 34 or so as another table `xi.campaign.leaders` or some such (I'm not great at naming things).

Reply:
If we collect the leaders and their associated mobs into a table like:
```lua
xi.campaign.leader =
{
    -- San d'Oria
    VALAINERAL_R_DAVILLES    = { 'Valaineral_R_Davilles',  'Royal_Guard'                                  },
    LEONOYNE                 = { 'Leonoyne',               'Savage_Hound_Condottiere'                     },
    ASHMEA_B_GREINNER        = { 'Ashmea_B_Greinner',      'Scarlet_Boar_Esquire'                         },
    FELDRAUTTE_I_ROUHENT     = { 'Feldrautte_I_Rouhent',   'Norvallen_Knight'                             },
    YRVAULAIR_S_COUSSERAUX   = { 'Yrvaulair_S_Cousseraux', 'Royal_Knight'                                 },
    -- Orcish
    CONQUEROR_BAKGODEK       = { 'Conqueror_Bakgodek',     'Steelhide Protector'                          },
    ONE_EYED_GWAJBOJ         = { 'One-eyed_Gwajboj',       'Gutrender Trooper'                            },
    DEATHLORD_ROJGNOJ        = { 'Deathlord_Rojgnoj',      'Spinebeak Neckchopper'                        },
    JAGIDBODS_WARMACHINE     = { 'Jagidbods_Warmachine',   'Jagidbod_of_Clan_Reaper', 'Clan_Reaper_Grunt' },
    ALPHA_GNOLE_ANDERS       = { 'Alpha_Gnole_Anders',     'Anderss_Guard',           'Moonfang_Warrior'  },
}
```
They could be assigned per area without duplication:
```lua
xi.campaign.area =
{
    [xi.zone.EAST_RONFAURE_S] =
    {
        [xi.campaign.army.SANDORIA] =
        {
            xi.campaign.leader.VALAINERAL_R_DAVILLES,
            xi.campaign.leader.LEONOYNE,
            xi.campaign.leader.ASHMEA_B_GREINNER,
            xi.campaign.leader.FELDRAUTTE_I_ROUHENT,
            xi.campaign.leader.YRVAULAIR_S_COUSSERAUX,
        },
        [xi.campaign.army.ORCISH] =
        {
            xi.campaign.leader.CONQUEROR_BAKGODEK,
            xi.campaign.leader.ONE_EYED_GWAJBOJ,
            xi.campaign.leader.DEATHLORD_ROJGNOJ,
            xi.campaign.leader.JAGIDBODS_WARMACHINE,
            xi.campaign.leader.ALPHA_GNOLE_ANDERS,
        },
    },
}
```
This would allow the additional area specific groups to be easily assigned. Leaders and their mobs can be selected out of the current area when Campaign begins.

Please let me know how you'd like me to proceed.

Reply:
some of them do exist in multiple zones so perhaps instead of keying by zone we could list whats zones they are allowed in and make sure we get the zone then `zonequeryEntitiesByName()` for that zone as needed? thinking out loud in text. maybe some very slight duplication is better than that, unsure. It wouldn't kill us to have one listed twice if need be.

Reply:
```lua
xi.campaign.area =
{
    {
        zones =
        {
            xi.zone.SOUTHERN_SAN_DORIA_S,
            xi.zone.EAST_RONFAURE_S,
            xi.zone.JUGNER_FOREST_S,
            xi.zone.LA_VAULE_S,
            xi.zone.BATALLIA_DOWNS_S,
            xi.zone.THE_ELDIEME_NECROPOLIS_S,
        },

        mobs  =
        {
            [xi.campaign.army.SANDORIA] =
            {
                VALAINERAL_R_DAVILLES    = { 'Valaineral_R_Davilles',  'Royal_Guard'                                    },
                LEONOYNE                 = { 'Leonoyne',               'Savage_Hound_Condottiere'                       },
                ASHMEA_B_GREINNER        = { 'Ashmea_B_Greinner',      'Scarlet_Boar_Esquire'                           },
                FELDRAUTTE_I_ROUHENT     = { 'Feldrautte_I_Rouhent',   'Norvallen_Knight'                               },
                YRVAULAIR_S_COUSSERAUX   = { 'Yrvaulair_S_Cousseraux', 'Royal_Knight'                                   },
                CERANE_I_VIRGAUT         = { 'Cerane_I_Virgaut',       'Aragoneu_Knight',         'Gold_Badger_Esquire' },
            },
            [xi.campaign.army.ORCISH] =
            {
                CONQUEROR_BAKGODEK       = { 'Conqueror_Bakgodek',     'Steelhide Protector'                            },
                ONE_EYED_GWAJBOJ         = { 'One-eyed_Gwajboj',       'Gutrender Trooper'                              },
                DEATHLORD_ROJGNOJ        = { 'Deathlord_Rojgnoj',      'Spinebeak Neckchopper'                          },
                JAGIDBODS_WARMACHINE     = { 'Jagidbods_Warmachine',   'Jagidbod_of_Clan_Reaper', 'Clan_Reaper_Grunt'   },
                ALPHA_GNOLE_ANDERS       = { 'Alpha_Gnole_Anders',     'Anderss_Guard',           'Moonfang_Warrior'    },
            },
        },
    },
}
```
This could be combined into a single table, featuring each of the three general areas (and eventually Northlands, which will include some overlap/duplication). This would require a small lookup with `utils.contains` without a direct mapping of zone ID to army (Could generate a mapping table on load, if it's allowed to have a simple loop outside of function).

Reply:
I dont think you need to specify which zones things are in, if you're going to be calling something like `xi.campaign.initArmies` in the startup of each zone that prepares all of the tables and IDs. After that it doesn't matter too much what the formula is, as long as it's pre-defined so IDEs can auto-complete it for later users.

Reply:
After discussing with Xaver we believe it's better to only submit the NPC offset and text ID as a starting point, so that the mob tables can be developed after determining the best course of action.

The design can be better suited once more of the functionality is properly understood and prototyped. Additionally, querying entities is going to require some missing capture data (Positions), so this can be submitted at the same time.

Reply:
neither of us here were expecting you to make any actual functionality beyond populating IDs in a future forward maintainable way